tests/cpp/merge_mps/merge_mps_test.cpp

```diff
@@ -0,0 +1,213 @@
+    }
+
+    void TearDown() override {
+    }
+
+    void createMasterProblem() {
+        std::ofstream master_problem(tmp_dir_ / "master.mps"s);
+        master_problem << R"(NAME          MASTER  FREE
+ROWS
+ N  OBJROW
+ L  C1
+ G  C2
+COLUMNS
+    X1        OBJROW       3.0   C1        2.0
+    X1        C2        1.0
+    X2        OBJROW       4.0   C1        1.0
+    X2        C2        2.0
+RHS
+    RHS      C1        8.0
+    RHS      C2        3.0
+BOUNDS
+ UP BOUND      X1        4.0
+ UP BOUND      X2        4.0
+ENDATA)";
+        master_problem.close();
+        options_.MASTER_NAME = "master.mps";
+    }
+
+    void createStructureFile(const std::vector<std::tuple<std::string, std::string, int> > &entries) {
+        std::ofstream structure_file(options_.STRUCTURE_FILE);
+        for (const auto &[mps, var, idx]: entries) {
+            structure_file << fmt::format("{0} {1} {2}\n", mps, var, idx);
+        }
+        structure_file.close();
+    }
+
+    std::filesystem::path tmp_dir_;
+    MergeMPSOptions options_;
+    std::shared_ptr<Xpansion::Test::LoggerNOOPStub> logger_;
+    std::shared_ptr<Xpansion::Test::InMemoryWriter> writer_;
+};
+
+TEST(MergeMPS, empty_input_ok) {
+    auto tmpDir = CreateRandomSubDir(std::filesystem::temp_directory_path());
+    std::ofstream structure_file(tmpDir / "structure_file.txt"s);
+
+    MergeMPSOptions options;
+    options.SOLVER_NAME = "COIN";
+    options.STRUCTURE_FILE = (tmpDir / "structure_file.txt"s).string();
+    auto logger = std::make_shared<Xpansion::Test::LoggerNOOPStub>();
+    auto writer = std::make_shared<Xpansion::Test::InMemoryWriter>();
+    std::filesystem::current_path(tmpDir);
+    MergeMPS mergeMPS(options, logger, writer);
+    mergeMPS.launch();
+    auto lastSolution = writer->solution_data_;
+    EXPECT_EQ(lastSolution.problem_status, "ERROR");
+}
+
+TEST_F(MergeMPSTest, one_master_Problem_ok) {
+    createMasterProblem();
+    createStructureFile({{"master.mps", "X1", 0}, {"master.mps", "X2", 1}});
+    std::filesystem::current_path(tmp_dir_);
+    MergeMPS mergeMPS(options_, logger_, writer_);
+    mergeMPS.launch();
+    auto lastSolution = writer_->solution_data_;
+    EXPECT_EQ(lastSolution.problem_status, "OPTIMAL");
+    //log_merged_mps exists
+    std::cout << tmp_dir_ / "log_merged.mps" << std::endl;
+    EXPECT_TRUE(std::filesystem::exists(tmp_dir_ / "log_merged.mps"s));
+    EXPECT_TRUE(std::filesystem::exists(tmp_dir_ / "log_merged.lp"s));
+    //Print the merged problem
+    std::ifstream merged_mps(tmp_dir_ / "log_merged.mps"s);
+    std::string line;
+    while (std::getline(merged_mps, line)) {
+        std::cout << line << std::endl;
+    }
+    merged_mps.close();
+    SolverFactory factory;
+    auto merged = factory.create_solver("CBC");
+    merged->read_prob_lp(tmp_dir_ / "log_merged.lp"s);
+    auto master = factory.create_solver("CBC");
+    master->read_prob_mps(tmp_dir_ / "master.mps"s);
+    //Merged and master problems are identical
```

Comment:
Here you just check that they have the same shape, not that they are identical (values could be different). To check equality between problems, either look at https://github.com/AntaresSimulatorTeam/antares-xpansion/blob/682a27571aea83bfc5a1bc4e922dfcb72c0c7051/tests/cpp/solvers_interface/test_modifying_problem.cpp#L7 or introduce a `==` operator in `SolverAbstract` ?